### DISCUSSION TITLE
have bug in fillable change diynamci

### DISCUSSION THREAD

**Discussion by vahidalvandi** [GeneaLabs/laravel-model-caching#486] [General]
**Describe the bug**
i need chnage fillable in eloquent by user role 

```
    public function __construct(array $attributes = []) {

        if(auth()->user()->is_admin){
            $this->fillable[] = 'is_active';
            $this->fillable[] = 'sales_discount';
            $this->fillable[] = 'business_package_id';
            $this->fillable[] = 'business_contract_id';
        }

        parent::__construct($attributes);

    }
``` 
but need twice update for get fillable   
**Environment**
 - PHP: 7.4
 - OS: Ubuntu 18.04 
 - Laravel: 6.18
 - Model Caching: 0.7.4

**Additional context**

**Comment by mikebronner** [comment]
Hi @vahidalvandi, Thanks for reporting this. My hunch says this issue is not related to the package. I will try to recreate this in a test case. If you are able to recreate this in a unit test, please submit a PR for it, that would help a lot.

**Comment by vahidalvandi** [comment]
Thank you  . This  bug related with this package because after remove cache
trait fix it

**Comment by vahidalvandi** [comment]
do it fixed?

**Comment by mikebronner** [comment]
@vahidalvandi Can you provide the eloquent query you are having problems with as well?
